server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -901,10 +925,54 @@ public boolean deletePool(DeletePoolCmd cmd) {
             s_logger.warn("Unable to delete storage id: " + id + " due to it is not in Maintenance state");
             throw new InvalidParameterValueException("Unable to delete storage due to it is not in Maintenance state, id: " + id);
         }
-        Pair<Long, Long> vlms = _volsDao.getCountAndTotalByPool(id);
+
+        if (sPool.getPoolType() == StoragePoolType.DatastoreCluster) {
+            // FR41 yet to handle on failure of deletion of any of the child storage pool
+            if (checkIfDataStoreClusterCanbeDeleted(sPool, forced)) {
```

Comment:
are we sure the check outside the transaction is secure/safe? even if it seems to be just the return, still in the check and during the stack drop context switches may have occurred. please reconsider.